src/agent/src/device.rs

```diff
@@ -56,9 +56,50 @@ struct DevIndexEntry {
 #[derive(Debug)]
 struct DevIndex(HashMap<String, DevIndexEntry>);
 
+fn filename_is_pci_addr(f: &std::ffi::OsStr) -> bool {
+    lazy_static! {
+        static ref PCIADDR: Regex =
+            Regex::new(r"^[0-9a-f]{4}:[0-9a-f]{2}:[0-9a-f][.][0-7]$").unwrap();
```

Comment:
`<paranoia>`I wonder if we should make this regex case-insensitive?`</paranoia>`.

Reply:
I think if the kernel were to go and change case in sysfs filenames, this would be the least of our worries.